ilp-core/src/test/java/org/interledger/core/InterledgerAddressTest.java

```diff
@@ -400,6 +402,28 @@ public void testToString() {
     assertThat(InterledgerAddress.of("g.foo").toString(), is("InterledgerAddress{value=g.foo}"));
   }
 
+  @Test
+  public void testGetInteractions() {
+    InterledgerAddress destinationAddress = InterledgerAddress.of("g.us-fed.ach.0.acmebank.swx0a0.acmecorp.sales.199.~ipr.cdfa5e16-e759-4ba3-88f6-8b9dc83c1868.2");
+    // @sappenin: Is the final match for "~ipr" or "ipr" as the first interaction?
```

Comment:
I added this question to the [forum post](https://forum.interledger.org/t/questions-about-interactions-in-ilp-addresses/717) as well. 

My understanding is that there's nothing normative about interactions - what we read in IL-RFC-15 is just an example, and implementations are free to represent interactions however they want.

To this end, we might just create some "Java ILP Address" rules of our own here. E.g., everything _after_ the first `~` is the interaction string. It can be de-composed by the `.` character.

In that world, the first interaction is `ipr`, but there should be an alternative method that provides access to the entire interaction string, which would be `~ipr.cdfa5e16-e759-4ba3-88f6-8b9dc83c1868.2`.

**That said**, let's see what comes out of the forum discussion before we move forward with this PR. There's a real possibility (one that I'm leaning towards) that interactions are just something that each consumer of this library can merely create on their own as a higher-order thing (not in this library). If we go that route, then this functionality doesn't belong inside of this class or library at all, and we can maybe just close this PR and #196.

Stay tuned...

Reply:
Perfect. Thank you for the detailed response and for creating the forum post. It'd be nice to hear the thoughts of the community on this RFC and have more clarity on the definition of interactions.

Reply:
I agree with the response in the forum. At this level of the library, I don't think we need to worry about "interactions". Sorry for the false issue in #196.